tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/lock/LocksInMainTest.java

```diff
@@ -134,6 +134,17 @@ public void testLockNegativeCases() {
                                   18, 9);
     }
 
+    @Test(description = "Test worker lock negative cases")
+    public void testWorkerSendReceiveInLockNegativeCases() {
+        CompileResult compileResult =
+                BCompileUtil.compile("test-src/lock/locks_in_worker_send_receive_negative.bal");
+        assertEquals(compileResult.getErrorCount(), 6);
```

Comment:
Capture all negative cases.

Reply:
Updated in 71cf6d05b3840513a959f848b8cf4d389670a66b